usr/lib/cca_stdll/cca_specific.c

```diff
@@ -1584,7 +1693,20 @@ static CK_RV cca_get_adapter_domain_selection_infos(STDLL_TokData_t *tokdata)
     long return_code, reason_code, rule_array_count, verb_data_length;
     unsigned int i;
     const char *val;
-
+#if defined(_AIX)
+    /*
+     * AIX does not have the luxury to identify the default card/domain
+     * combination that is assigned to an LPAR/host - this data is exposed
+     * through sysfs, but only in case of Linux on Z. Because of this, we must
+     * lean on the host library to decide which card to use when there is no
+     * user preference - the default card we use is the one that is the first
+     * entry written to the data buffer by the host library.
+     *
+     * That's why we force ANY device/domain usage - this will be overwritten 
+     */
+     cca_private->dev_any = FALSE;
+     cca_private->dom_any = FALSE;
+#endif
```

Comment:
I guess you wanted to use ifdef with else to omit the getenv calls in case of AIX.
Setting dev_any and dom_any to FALSE is superfluous, they are initialized to false already.
What you really want is that they are never set to TRUE on AIX. So I guess you should ifdef out the block below containing the getenv's for AIX.